### PR TITLE
Update main table colors

### PR DESCRIPTION
**Pull request by LoayGhreeb**
Follow-up to: #11510. Closes https://github.com/LoayGhreeb/jabref/issues/8.
Before:
image

After:
image

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/Base.css [hunk 1]
```diff
@@ -649,15 +649,27 @@ TextFlow > .tooltip-text-monospaced {
 }
 
 .table-row-cell:matching-search-and-groups {
-    -fx-background-color: white;
+    -fx-background-color: #FFFFFFFF;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/Base.css:652]: Please either make them derived or add some new constants, as this makes it easier for dark theme to change the colors

**Comment by koppor** [comment]
Carl and I are on it.

**Comment by koppor** [comment]
Using this gray-blue colors looks odd:

image

**Comment by koppor** [comment]
Blue berry:

image

**Comment by koppor** [comment]
With `-jr-theme`:

image

**Comment by koppor** [comment]
With -jr-theme, memories of the C64 come back. I "strictly" used `-jr-theme` color 😇

**Comment by koppor** [comment]
I added `src/test/resources/testbib/simple-search-library.bib` to be able to display all 4 cases on a single page.

- Search for `S`
- Selectd group `g`

Then, `Sg` matches both, `S` only the search string, `g` only group`, and `n` none of them.

**Review by calixtus — DISMISSED** [review]

**Comment by LoayGhreeb** [comment]
Double-clicking on an even row removes the selection color

**Comment by LoayGhreeb** [comment]
The text is not shown on the selected non-matched entry.
image

**Comment by koppor** [comment]
> However, if there is an active search or a selected group, all rows will lose the selection color when double-clicked.

Single click works. What magic does JabRef do on a double click?

**Comment by LoayGhreeb** [comment]
The only action on double-click is to open the entry editor: https://github.com/JabRef/jabref/blob/db9f83cf77b1a5215e729d3c336371b4edef04a4/src/main/java/org/jabref/gui/maintable/MainTable.java#L142-L146

**Comment by koppor** [comment]
If we cannot fix the next 20 hours, we merge and create two follow-up issues. - Without stripes, it looks really bad IMHO.

**Comment by LoayGhreeb** [comment]
Sorry for the confusion in https://github.com/JabRef/jabref/pull/11579#issuecomment-2270047007, but the issue only affects the even rows. Odd rows keep the selection color in all cases.

**Comment by koppor** [comment]
I cannot manage to "double filter":

```css
.table-row-cell:not(.focused):matching-groups-not-search > .table-cell {
    -fx-text-fill: -jr-match-3-text-color;
}
```

That would be requried to have the text color only applied when not focused - but does not trigger at all.

Seems to be  known issue - https://stackoverflow.com/q/39355081/873282

Thus, added a work around at [`e7c0460` (#11579)](https://github.com/JabRef/jabref/pull/11579/commits/e7c0460f1db8d59c1ffffa5fa37449220fdc8973).

**Comment by koppor** [comment]
The mentioned issues should be fixed now.

**Review by calixtus — APPROVED** [review]

**Comment by koppor** [comment]
Other color scheme PR: https://github.com/JabRef/jabref/pull/3839

**Pull request merged**